src/shipit_code_coverage_backend/shipit_code_coverage_backend/coverage.py

```diff
@@ -137,21 +140,236 @@ def _get(url=''):
         return await get_mercurial_commit(commit['commitid']), await get_mercurial_commit(commit['parent'])
 
 
+class ActiveDataClient():
+    '''
+    Active data async client, through Elastic Search
+    '''
+    def __init__(self):
+        self.client = AsyncElasticsearch(hosts=secrets.ACTIVE_DATA_HOSTS)
+
+    async def __aenter__(self):
+        # TODO: Should we ping the server here (overkill ?)
+        return self.client
+
+    async def __aexit__(self, *args, **kwargs):
+
+        # TODO: Find out why this line fails on
+        # RuntimeWarning: coroutine 'ClientSession.close' was never awaited
+        # It should remove the warning about unclosed connector
+        pass
+
+
 class ActiveDataCoverage(Coverage):
-    URL = 'https://activedata.allizom.org/query'
+
+    FIELD_REPO = 'repo.branch.name.~s~'
+    FIELD_FILENAME = 'source.file.name.~s~'
+    FIELD_CHANGESET = 'repo.changeset.id.~s~'
+    FIELD_CHANGESET_DATE = 'repo.changeset.date.~n~'
+    FIELD_TOTAL_PERCENT = 'source.file.percentage_covered.~n~'
+
+    @staticmethod
+    async def list_tests(changeset, filename):
+        '''
+        List all the tests available for file in a changeset
+        '''
+        filters = [
+            # Always query on Mozilla Central
+            {'term': {ActiveDataCoverage.FIELD_REPO: 'mozilla-central'}},
```

Comment:
We'll need a few additional default filters to match what we're currently doing:
1) Ignore awsy and talos suites (they are not test suites, so their coverage doesn't have the same value);
2) Ignore obj-firefox/* (these files are generated files, we don't care as much if they are covered or not).